### PR TITLE
Add whanau (fixes #84)

### LINKED ISSUE #84: As administator I need to create/invite user to wharehauora by email
- [ ] User with "janitor" role can create a user
- [ ] user receives an email inviting them to WhareHauora
- [ ] user can create a home, or admin can create the home when they invite the user
- [ ] admin an provision MQTT for the home (existing functionality)

### PR DESCRIPTION
**Pull request by fauxparse**
This PR adds the concept of an invitation:

* A user invites a whānau member by entering their email address.
* An invitation with a unique and unguessable token is created.
* The invitee receives an email with a link to accept (or decline) the invitation.
* If this is their first time using the application, they are prompted to log in or sign up.
* They are shown a screen with buttons to accept or decline.
* If they accept, they are taken to the whare screen.
* If they decline, they are taken to their dashboard.

If the invitee does not wish to accept the invitation, no new user record is created. This ensures that only people who have given permission have their email addresses stored. This model also allows users to sign up with an email address that may be more convenient than the one with which they were invited.

Users can see a list of outstanding invitations on their whānau page, and can delete any they wish to rescind.

### CODE REVIEW

File: app/controllers/invitations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
```
Comment (Br3nda) [inline app/controllers/invitations_controller.rb]: :+1:

File: app/views/invitation_mailer/invitation_email.text.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Kia ora,
```
Comment (Br3nda) [inline app/views/invitation_mailer/invitation_email.text.erb]: If you have time, could this be haml like the rest of the app?

File: spec/controllers/invitations_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        expect(invitation).to be_present
+        expect do
+          delete :destroy, home_id: home.to_param, id: invitation.to_param
+        end.to change { home.invitations.count }.by(-1)
+        expect(response).to redirect_to(home_home_viewers_path(home))
+      end
+    end
+  end
+
+  # context 'signed in as admin/janitor' do
```
Comment (Br3nda) [inline spec/controllers/invitations_controller_spec.rb]: Is this something you're hoping to fill in later?

Comment (fauxparse) [inline spec/controllers/invitations_controller_spec.rb]: Ohhhh, good catch.

File: app/views/invitation_mailer/invitation_email.text.erb (hunk 2 above)
Comment (fauxparse) [inline app/views/invitation_mailer/invitation_email.text.erb]: I wasn't sure about plain text in HAML? I didn't think it was a thing.

**Comment by Br3nda** [comment]
Awesome! Thanks @fauxparse

**Pull request merged**